src/fieldAnalyser/typeUtils.ts

```diff
@@ -0,0 +1,38 @@
+import {FieldTypes} from '@coveord/platform-client';
+
+const acceptedTransitions: Array<{from: FieldTypes; to: Array<FieldTypes>}> = [
+  {from: FieldTypes.LONG, to: [FieldTypes.LONG_64, FieldTypes.DOUBLE]},
+  {from: FieldTypes.LONG_64, to: [FieldTypes.DOUBLE]},
+];
+
+export function isValidTypeTransition(
+  currentState: FieldTypes,
+  nextState: FieldTypes
+): boolean {
+  const sameStateTransition = currentState === nextState;
+  const differentStateTransition = acceptedTransitions
+    .find((a) => a.from === currentState)
+    ?.to.includes(nextState);
+  return sameStateTransition || Boolean(differentStateTransition);
```

Comment:
```suggestion
  if(currentState === nextState) {
    return true;
  }
  const differentStateTransition = acceptedTransitions
    .find((a) => a.from === currentState)
    ?.to.includes(nextState);
  return Boolean(differentStateTransition);
```

Will allow skipping the array shenanigans when possible.